buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -749,17 +760,33 @@ protected long unsafeOffset() {
             // Constructor only used by the MAGAZINE_FREED sentinel.
             delegate = null;
             magazine = null;
+            allocator = null;
             capacity = 0;
             pooled = false;
         }
 
         Chunk(AbstractByteBuf delegate, Magazine magazine, boolean pooled) {
             this.delegate = delegate;
-            this.magazine = magazine;
             this.pooled = pooled;
             capacity = delegate.capacity();
-            magazine.usedMemory.getAndAdd(capacity);
             updater.setInitialValue(this);
+            allocator = magazine.parent;
+            attachToMagazine(magazine);
+        }
+
+        Magazine currentMagazine()  {
+            return magazine;
+        }
+
+        void detachFromMagazine() {
+            magazine.usedMemory.getAndAdd(-capacity);
```

Comment:
Only do this if `magazine` is not already `null`. This can happen through a line of calls that start at `freeCentralQueue()`.